src/stories/Blocks/header/Header.tsx

```diff
@@ -114,6 +114,7 @@ export const Header = (props: HeaderProps) => {
                   "header__menu-dropdown-icon--expanded": isDropdownOpen,
                 })}
                 onClick={() => setIsDropdownOpen(!isDropdownOpen)}
```

Comment:
The markup should be the same as in Drupal. Remove the onClick and wrap the element in a button ☺️

Reply:
I have changed the markup to match Drupal. Button element now contains the onClick and is wrapping the icon.